include/semaphore.h

```diff
@@ -104,8 +104,16 @@ struct semholder_s
 
 struct sem_s
 {
-  volatile int32_t semcount;     /* >0 -> Num counts available */
```

Comment:
since we always use macro access semcount/mholder, why not remove the union?

Reply:
Could do, I was thinking that it is safer / cleaner to do it like this. What would be a good name for combined holder/semcount?